sentry_sdk/integrations/spark/spark_driver.py

```diff
@@ -0,0 +1,174 @@
+from sentry_sdk import configure_scope
+from sentry_sdk.hub import Hub
+from sentry_sdk.integrations import Integration
+
+
+class SparkIntegration(Integration):
+    identifier = "spark"
+
+    @staticmethod
+    def setup_once():
+        # type: () -> None
+        patch_spark_context_init()
+
+
+def patch_spark_context_init():
+    # type: () -> None
+    from pyspark import SparkContext  # type: ignore
+    from pyspark.java_gateway import ensure_callback_server_started  # type: ignore
+
+    spark_context_init = SparkContext._do_init
+
+    def _sentry_patched_spark_context_init(self, *args, **kwargs):
+        init = spark_context_init(self, *args, **kwargs)
+
+        if not getattr(self, "_sentry_is_patched", False):
```

Comment:
Is the concern that `_do_init` is called multiple times? Are there possibilities that the method is called concurrently?

Reply:
The main concern was that _do_init could be called multiple times, and I did not want to open a py4j gateway and call `addSparkListener` multiple times